apps/src/storage/dataBrowser/data-styles.module.scss

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file Styles shared by various react components in the applab data browser.
+ */
+
+@import 'color.scss';
+
+$rowHeight: 45px;
+$cellPadding: 10px;
+$editButtonCellWidth: 142px;
+$maxCellWidth: 350px;
+
+.row {
+    height: $rowHeight;
+}
+
+.cell {
+    padding: $cellPadding;
+    border: 1px solid gray;
+    font-size: 14px;
+    max-width: $maxCellWidth;
+    overflow: hidden;
+    &AddButton {
+        background-color: $lightest_teal
+    }
+    &EditButton {
+        background-color: $lightest_teal;
+        width: $editButtonCellWidth
+    }
+}
+
+.headerCell {
+    background-color: $teal;
+    border: 1px solid gray;
+    max-width: $maxCellWidth;
+    padding: 6px 10px;
+    white-space: nowrap;
+}
+
+.link {
+    color: $purple;
+    cursor: pointer;
+    font-family: "Gotham 7r", sans-serif
+}
+
+.button {
+    display: inline-block;
+    font-size: 14px;
+    line-height: 20px;
+    margin: 0px;
+    padding: 4px 12px;
+    text-align: center;
+    vertical-align: middle;
+
+    &White {
+        background-color: white;
+        color: black;
+    }
+
+    &Red {
+        background-color: $bootstrap_button_red;
+        color: white;
+    }
+
+    &Blue {
+        background-color: $bootstrap_button_blue;
+        color: white;
+    }
+
+    &Edit {
+        margin-right: 10px;
+        min-width: 60px;
+    }
+
+    &Save {
+        margin-right: 10px;
+        min-width: 60px;
+    }
+
+    &Gray {
+        margin-left: 5px;
+        background-color: $background_gray;
+    }
+}
+
+.input {
+    width: calc(100% - 14px);
+    height: 20px;
+    border: 1px solid gray;
+    border-radius: 5px;
+    padding: 4px 6px;
+}
+
+.viewHeader {
+    margin-top: 10px;
+    margin-bottom: 10px;
+    margin-left: 0px;
+    margin-right: 0px;
+}
+
+.backLink {
+    font-size: 24px;
+}
+
+.debugLink {
+    float: right;
+}
```

Comment:
these styles should either be colocated with `.link` or nested within them -- if they are completely unrelated, i wouldn't nest them, but otherwise i'd nest them (which would probably change the classnames to "linkBack" and "linkDebug")

Reply:
They're _sort of_ related? `.backLink` and `.debugLink` are classes used for span elements that contain the actual links. The span elements should not follow the rules for the `.link` class. That's why I kept them separate, but I think it would still work to nest and rename.

```
        <div className={dataStyles.viewHeader}>
          <span className={dataStyles.backLink}>
            <a
              id="tableBackToOverview"
              className={dataStyles.link}
              onClick={() => onViewChange(DataView.OVERVIEW)}
            >
              <FontAwesome icon="arrow-circle-left" />
              &nbsp;Back to data
            </a>
          </span>
          <span className={dataStyles.debugLink}>
            <a
              id="uitest-tableDebugLink"
              className={dataStyles.link}
              onClick={this.toggleDebugView}
            >
              {this.state.showDebugView ? 'Table view' : 'Debug view'}
            </a>
          </span>
        </div>
```

As a side note, I believe these only are used in `DataTableView.jsx`, so many they should move to that module rather than this shared on.

Reply:
oh, good call -- if these styles are specific to that component, we should move them to that module

Reply:
https://github.com/code-dot-org/code-dot-org/pull/47567/commits/decbc7283ddffe7bda5b19c956c868b272247bf3